src/Psl/DateTime/Timestamp.php

```diff
@@ -0,0 +1,266 @@
+     *
+     * @return int Seconds since the epoch. Can be negative for times before the epoch.
+     *
+     * @mutation-free
+     */
+    public function getSeconds(): int
+    {
+        return $this->seconds;
+    }
+
+    /**
+     * Returns the nanoseconds part of this timestamp.
+     *
+     * @return int<0, 999999999> The nanoseconds part, ranging from 0 to 999999999.
+     *
+     * @mutation-free
+     */
+    public function getNanoseconds(): int
+    {
+        return $this->nanoseconds;
+    }
+
+    /**
+     * Adds the specified duration to this timestamp object, returning a new instance with the added duration.
+     *
+     * @throws Exception\UnderflowException If adding the duration results in an arithmetic underflow.
+     * @throws Exception\OverflowException If adding the duration results in an arithmetic overflow.
+     *
+     * @mutation-free
+     */
+    public function plus(Duration $duration): static
+    {
+        [$h, $m, $s, $ns] = $duration->getParts();
+        $totalSeconds = SECONDS_PER_MINUTE * $m + SECONDS_PER_HOUR * $h + $s;
+        $newSeconds = $this->seconds + $totalSeconds;
+        $newNanoseconds = $this->nanoseconds + $ns;
+
+        // No manual normalization required here due to fromRaw handling it
+        return self::fromRaw($newSeconds, $newNanoseconds);
+    }
+
+    /**
+     * Subtracts the specified duration from this timestamp object, returning a new instance with the subtracted duration.
+     *
+     * @throws Exception\UnderflowException If subtracting the duration results in an arithmetic underflow.
+     * @throws Exception\OverflowException If subtracting the duration results in an arithmetic overflow.
+     *
+     * @mutation-free
+     */
+    public function minus(Duration $duration): static
+    {
+        [$h, $m, $s, $ns] = $duration->getParts();
+        $totalSeconds = SECONDS_PER_MINUTE * $m + SECONDS_PER_HOUR * $h + $s;
+        $newSeconds = $this->seconds - $totalSeconds;
+        $newNanoseconds = $this->nanoseconds - $ns;
+
+        // No manual normalization required here due to fromRaw handling it
+        return self::fromRaw($newSeconds, $newNanoseconds);
+    }
+
+    /**
+     * Formats the timestamp according to the given format and timezone.
+     *
+     * The format can be a predefined {@see DateFormat} case, a date-time format string, or `null` for default formatting.
+     *
+     * The timezone is required to accurately represent the timestamp in the desired locale.
+     *
+     * @mutation-free
+     */
+    public function format(Timezone $timezone, null|DateFormat|string $format = null, ?Locale\Locale $locale = null): string
+    {
+        return Internal\zone_override($timezone, function () use ($timezone, $locale, $format): string {
+            $obj = new NativeDateTime();
+            $obj->setTimestamp($this->getSeconds());
+            $obj->setTimezone(new NativeDateTimeZone($timezone->value));
```

Comment:
I've seen you using the zone_override a couple of times for this and it seems a bit odd.
Wouldn't it be possible to set the timezone through the constructor instead of relying on the system's timezone?

Reply:
replaced by IntlTimezone.